src/blockchain_utilities/blockchain_find_locked.cpp

```diff
@@ -0,0 +1,469 @@
+  }
+};
+
+const std::string report_slot_entry_header = "num standard locks opened,num nonstandard blocktime locks opened,"
+  "num nonstandard unixtime locks opened,num total locks opened,num standard locks closed,"
+  "num nonstandard blocktime locks closed,num nonstandard unixtime locks closed,num total locks closed,"
+  "standard locks balance,nonstandard blocktime locks balance,nonstandard unixtime locks balance,total locks balance";
+
+std::ostream& operator<<(std::ostream &os, const report_slot_entry_t &report_slot_entry)
+{
+  os << report_slot_entry.num_standard_locks_opened;
+  os << ',';
+  os << report_slot_entry.num_nonstandard_blocktime_locks_opened;
+  os << ',';
+  os << report_slot_entry.num_nonstandard_unixtime_locks_opened;
+  os << ',';
+  os << report_slot_entry.num_total_locks_opened;
+  os << ',';
+
+  os << report_slot_entry.num_standard_locks_closed;
+  os << ',';
+  os << report_slot_entry.num_nonstandard_blocktime_locks_closed;
+  os << ',';
+  os << report_slot_entry.num_nonstandard_unixtime_locks_closed;
+  os << ',';
+  os << report_slot_entry.num_total_locks_closed;
+  os << ',';
+
+  os << report_slot_entry.standard_locks_balance;
+  os << ',';
+  os << report_slot_entry.nonstandard_blocktime_locks_balance;
+  os << ',';
+  os << report_slot_entry.nonstandard_unixtime_locks_balance;
+  os << ',';
+  os << report_slot_entry.total_locks_balance;
+  //os << ',';
+
+  return os;
+}
+
+static uint32_t get_first_spendable_block_index(
+  const uint32_t unlock_time,
+  const uint32_t block_included_in_chain)
+{
+  uint32_t unlock_block_index = 0;
+
+  static_assert(CRYPTONOTE_DEFAULT_TX_SPENDABLE_AGE > 0, "unexpected default spendable age");
+  const uint32_t default_block_index = block_included_in_chain + CRYPTONOTE_DEFAULT_TX_SPENDABLE_AGE;
+
+  if (unlock_time < CRYPTONOTE_MAX_BLOCK_NUMBER)
+  {
+    unlock_block_index = unlock_time;
+  }
+  else // unix timelock
+  {
+    // Interpret the unlock_time as time
+    // TODO: hardcode correct times for each network and take in nettype
+    const uint32_t hf_v15_time = 1656629118;
+    const uint32_t hf_v15_height = 2689608;
+
+    // Use the last hard fork's time and block combo to convert the time-based timelock into an unlock block
+    // TODO: consider taking into account 60s block times when that was consensus
+    if (hf_v15_time > unlock_time)
+    {
+      const uint32_t seconds_since_unlock = hf_v15_time - unlock_time;
+      const uint32_t blocks_since_unlock = seconds_since_unlock / DIFFICULTY_TARGET_V2;
+
+      unlock_block_index = hf_v15_height > blocks_since_unlock
+        ? (hf_v15_height - blocks_since_unlock)
+        : 0;
+    }
+    else
+    {
+      const uint32_t seconds_until_unlock = unlock_time - hf_v15_time;
+      const uint32_t blocks_until_unlock = seconds_until_unlock / DIFFICULTY_TARGET_V2;
+      unlock_block_index = hf_v15_height + blocks_until_unlock;
+    }
```

Comment:
Thank you for this :)

Does this end up producing reasonable / expected values? This seemed the most sane approach to me but I didn't test it. Some analysis of the results this function in particular is producing would be nice (focusing on block timestamp reasonability would be ok imo). Making sure this is correct is on the general TODO list for the fcmp++ integration

It's definitely wrong for any timestamp locked outputs created before 120s block times, so identifying the impact there would also be helpful (making sure all of those should all already be unlocked would be ok e.g.).

Overall likely shouldn't affect the results you highlighted, but relevant for shipping this code